File: kandoe_service_card/src/Models/Models/Cards/Card.cs
using System;
using System.Collections.Generic;
using MongoDB.Bson.Serialization.Attributes;

namespace Models.Models.Cards
{
    public class Card : ICard
    {
        [BsonId]
        public string CardId { get; set; }
        public string ImageUrl { get; set; }
        public string Text { get; set; }
        public string ThemeId { get; set; }
        public DateTime UpdatedOn { get; set; } = DateTime.Now;
        public DateTime CreatedOn { get; set; } = DateTime.Now;
        public bool IsEnabled { get; set; } = true;
        public ICollection<Reaction> Reactions { get ; set ; }

        public Card()
        {
            CardId = Guid.NewGuid().ToString();
            CreatedOn = DateTime.UtcNow;
            IsEnabled = true;
            Reactions = new List<Reaction>();
        }
    }
}

File: kandoe_angular2/src/app/domain/card.ts
export interface Card {
  cardId: string;
  imageUrl: string;
  text: string;
  themeId: string;
  reactions?: CardReaction[];
  updatedOn?: string;
  createdOn?: string;
  isEnabled?: boolean;
  // Enkel client-side
  selected?: boolean;
}

export interface CardReaction{
  cardId: string;
  username: string;
  message: string;
  time?: string;
}
